libs/components/modals/testing/src/modal/modal-harness.ts

```diff
@@ -83,4 +83,15 @@ export class SkyModalHarness extends SkyComponentHarness {
     const modal = this.#getModal();
     return (await modal).hasClass('sky-modal-full-page');
   }
+
+  /**
+   * Whether the modal has {@link SkyModalIsDirtyDirective.isDirty} set to dirty.
+   */
+  public async isDirty(): Promise<boolean> {
```

Comment:
One of the goals with these types of features is to keep developers from different teams from rewriting the same code. My hope is for that to apply to tests as well. I added this so that a developer could easily unit test that they were setting `isDirty` and that it contained the expected value. Without this, I'm not sure how developers would be able to test this without seeing if a confirmation modal opened.